extras/bindings/c/src/SolverInterfaceC.cpp

```diff
@@ -58,8 +61,11 @@ double precicec_advance(double computedTimestepLength)
 
 void precicec_finalize()
 {
-  PRECICE_ASSERT(interface != nullptr);
-  interface->finalize();
+  if (interface != nullptr)
+  {
+    interface->finalize();
+    interface.reset();
+  }
```

Comment:
This is not consistent with the Fortran bindings:

https://github.com/precice/precice/blob/583339a950427b530bc4872649145e14f66cb0ea/extras/bindings/fortran/src/SolverInterfaceFortran.cpp#L70-L75

A `PRECICE_CHECK` should suffice.

Reply:
I have changed that to be consistent with the Fortran interface in the sense that there is a `PRECICE_ASSERT` before finalizing preCICE and resetting the `unique_ptr` in 30d69b59149f9073170ddd30884ae7569ca72d39.